test/broadcast.jl

```diff
@@ -256,3 +255,10 @@ end
 let foo = [[1,2,3],[4,5,6],[7,8,9]]
     @test max.(foo...) == broadcast(max, foo...) == [7,8,9]
 end
+
+# Issue 17314
```

Comment:
And `broadcast(x -> x < 0 ? false : x, Int[])` should probably return the inferred type `Union{Bool,Int}[]`.